lib/opencensus/trace.ex

```diff
@@ -1,6 +1,25 @@
 defmodule Opencensus.Trace do
   @doc "Wraps the given block in a tracing child span with the given label/name and optional attributes"
   defmacro with_child_span(label, attributes \\ quote(do: %{}), do: block) do
+    quote do
+      parent_span_ctx = :ocp.current_span_ctx()
+      new_span_ctx = :oc_trace.start_span(unquote(label), parent_span_ctx, unquote(attributes))
+      :ocp.with_span_ctx(new_span_ctx)
+
+      try do
+        unquote(block)
+      after
+        :oc_trace.finish_span(new_span_ctx)
+        :ocp.with_span_ctx(parent_span_ctx)
+      end
+    end
+  end
+
+  @doc """
+  Wraps the given block in a tracing child span with the given label/name, optional attributes and
+  includes default attributes of line, module, file and function name in the span data.
+  """
+  defmacro with_child_span_defaults(label, attributes \\ quote(do: %{}), do: block) do
```

Comment:
I am not sure about distinction. I would rather allow setting `attributes` to `%{file: nil}` to prevent it from sending to the reporter. Alternatively we could make default attributes default and `with_child_span_naked/2-3` macro for handling it **without** defaults.